irods_query.hpp

```diff
@@ -203,6 +219,24 @@ namespace irods {
         class spec_query_impl : public query_impl_base {
             public:
             virtual ~spec_query_impl() {
+                // Close all statements for this query
+                spec_input_.continueInx = gen_output_->continueInx + 1;
+                freeGenQueryOut(&gen_output_);
+                spec_input_.maxRows = 0;
+                while (spec_input_.continueInx > 0) {
+                    auto err = query_helper::spec_query_fcn(
+                                   comm_,
+                                   &spec_input_,
+                                   &gen_output_);
+                    if (err < 0) {
+                        irods::log(ERROR(
+                                    err, (boost::format(
+                                    "[%s] - Failed to close statement with continueInx [%d]") %
```

Comment:
mention this is gen vs spec in the error message?

Reply:
Done